ambry-store/src/main/java/com.github.ambry.store/CompactionLog.java

```diff
@@ -336,21 +350,23 @@ Phase getPhase() {
 
         compactionDetails (see CompactionDetails#toBytes())
         copyStartTime
-        switchStartTime
+        commitStartTime
         cleanupStartTime
         cycleEndTime
         safeToken (see StoreFindToken#toBytes())
        */
       byte[] compactionDetailsBytes = compactionDetails.toBytes();
-      byte[] safeTokenBytes = safeToken.toBytes();
-      int size = compactionDetailsBytes.length + 4 * TIMESTAMP_SIZE + safeTokenBytes.length;
+      byte[] safeTokenBytes = safeToken != null ? safeToken.toBytes() : ZERO_LENGTH_ARRAY;
```

Comment:
Same as the discussion [here](https://github.com/linkedin/ambry/pull/537#discussion_r99874334).